test/node/pipe.js

```diff
@@ -92,4 +100,48 @@ describe("request pipe", () => {
     });
     req.pipe(stream);
   });
+
+  it("should follow redirects", done => {
+    const stream = fs.createWriteStream(destPath);
+
+    let responseCalled = false;
+    const req = request.get(base + '/redirect');
+    req.type("json");
+
+    req.on("response", res => {
+      res.status.should.eql(200);
+      responseCalled = true;
+    });
+    stream.on("finish", () => {
+      JSON.parse(fs.readFileSync(destPath, "utf8")).should.eql({
+        name: "tobi",
+      });
+      responseCalled.should.be.true();
+      done();
+    });
+    req.pipe(stream);
+  });
+
+  it("should not throw on bad redirects", done => {
```

Comment:
added a test to prove that redirects with a missing Location will not throw and error.